src/arrayserialize.rs

```diff
@@ -46,13 +52,17 @@ impl<'a, A, D> serde::ser::MapVisitor for AVisitor<'a, A, D>
     {
         match self.state {
             0 => {
-                self.state += 1;
-                Ok(Some(try!(serializer.visit_map_elt("shape", self.arr.dim()))))
-            }
+                self.state +=1;
+                Ok(Some(try!(serializer.serialize_map_elt("v", ARRAY_FORMAT_VERSION))))
+            },
             1 => {
                 self.state += 1;
-                Ok(Some(try!(serializer.visit_map_elt("data", self.arr.iter()))))
-            }
+                Ok(Some(try!(serializer.serialize_map_elt("dim", self.arr.dim()))))
+            },
+            2 => {
+                self.state += 1;
+                Ok(Some(try!(serializer.serialize_map_elt("data", self.arr.iter()))))
```

Comment:
Are you thinking of `serialize_struct_elt`? Replacing `serializer.serialize_map_elt("data", self.arr.iter()` with `serializer.serialize_struct_elt("data", self.arr)` leads to:

````
thread 'serial_serde' has overflowed its stack
fatal runtime error: stack overflow
Process didn't exit successfully: `/home/janis/github/rust-ndarray/target/debug/serialize-0fbcfd722bad2949 serial --nocapture` (signal: 4, SIGILL: illegal instruction)
````